### PR TITLE
Improvement/5 3/button actions

### PR DESCRIPTION
**Pull request by chfsx**
Hi @Amstutz 
As discussed some weeks ago, this PR changes all buttons in the UI framework from <a> tags to <button> tags using onlick-actions.

Feel free to give any feedback, I'll try to implement your expertise.

Best regards, Fabian

### CODE REVIEW

File: src/UI/Implementation/Component/Button/Renderer.php [hunk 1]
```diff
@@ -45,18 +50,24 @@ protected function renderButton(Component\Button\Button $component, RendererInte
 		}
 
 		$tpl = $this->getTemplate($tpl_name, true, true);
+
 		$action = $component->getAction();
-		// The action is always put in the data-action attribute to have it available
-		// on the client side, even if it is not available on rendering.
-		$tpl->setVariable("ACTION", $action);
+		if ($action) {
```
Comment (klees) [inline src/UI/Implementation/Component/Button/Renderer.php]: For my curiosity: This seems to imply that the invariant that buttons have actions is broken somehow for the types of buttons that are rendered here. How?

Comment (chfsx) [reply src/UI/Implementation/Component/Button/Renderer.php]: oh, your right, actions are mandatory. sorry, just changed back that

**Comment by Amstutz** [comment]
@chfsx thanks a lot for this.

Looking good to me. However, final green light is need from @klees on this one, since he made the initial button implementation.

**Review by Amstutz — APPROVED** [review]

**Comment by klees** [comment]
It seems as if there are changes in here that don't belong to this PR?

**Comment by chfsx** [comment]
definitely... no idea where they're coming from. I'll adjust the PR

**Review by klees — APPROVED** [review]
Buttons be buttons. I'm fine with the changes code-wise. Besides that the change is invisible to users, right?

**Comment by chfsx** [comment]
Then it was "wrong" before my changes, since if you passed an empty string as action before, the href attribute was filled with this empty string, too. I has an `if($action)` in the PR first but changed it back after the comment from @klees. Shall I propose a PR with the change that the Signal is only registered when passing an non empty action?

**Comment by chfsx** [comment]
there you go: https://github.com/ILIAS-eLearning/ILIAS/pull/794

**Comment by Amstutz** [comment]
Why wrong? Well maybe from purist standpoint, but I guess that not your angle here. Note that previously we worked with href, which did not fire if left completely empty. Maybe not the best way to go, but this was how it behaved, so forcing to pass an # is an (implicit) change of interface. I can live with both options.

**Comment by chfsx** [comment]
I'm quite sure that my browser fires empty `hrefs` as well which results in a page reload, let me check that briefly

**Comment by chfsx** [comment]
I just checked in Firefox but a `<a href=''>TESTLINK</a>`leads to a page reload.

**Comment by Amstutz** [comment]
> I just checked in Firefox but a <a href=''>TESTLINK</a>leads to a page reload.

That's not the point here. The question is, wheter the event propagation is interrupted somewhere by the event firing the modal. I thought we had a switch which did that. I was wrong "if left completely empty" was utter nonsense. It's much rather, "if any event is attached". The propagation is stopped at (AbstractComponentRenderer):

```
$('#{$id}').on('{$event}', function(event) {
						$(this).trigger('{$signal}',
							{
								'id' : '{$signal}', 'event' : '{$event}',
								'triggerer' : $(this),
								'options' : JSON.parse('{$options}')
							}
						);
					return false; //Here the propagation is stopped forcefully, no matter what 
					});
```
This does not work anymore, since you are not in the same chain of event but register a second one. So we need a new solution as the one proposed in: https://github.com/ILIAS-eLearning/ILIAS/pull/794. As said, I can live with both option. @chfsx what do you prefer?

**Pull request merged**